pages/c/_cluster/fleet/index.vue

```diff
@@ -118,29 +119,92 @@ export default {
       });
     },
     badgeClass(area, row) {
+      // classes are defined in the themes SASS files...
       switch (area) {
       case 'clusters':
         if (row.clusterInfo?.ready === row.clusterInfo?.total && row.clusterInfo?.ready) {
-          return 'green-badge';
+          return `bg-${ STATES[STATES_ENUM.ACTIVE].color }`;
         }
 
-        return 'red-badge';
+        return `bg-${ STATES[STATES_ENUM.NOT_READY].color } badge-class-default`;
       case 'bundles':
-        if (row.bundlesReady?.length === row.bundles.length && row.bundlesReady?.length) {
-          return 'green-badge';
+        if (row.bundles?.length && row.bundles?.every(bundle => bundle.state?.toLowerCase() === STATES_ENUM.ACTIVE)) {
+          return STATES[STATES_ENUM.ACTIVE].color ? `bg-${ STATES[STATES_ENUM.ACTIVE].color }` : `bg-${ STATES[STATES_ENUM.UNKNOWN].color } bg-unmapped-state`;
+        }
+        if (row.bundles?.length && row.bundles?.some(bundle => bundle.state?.toLowerCase() === STATES_ENUM.ERR_APPLIED)) {
+          return STATES[STATES_ENUM.ERR_APPLIED].color ? `bg-${ STATES[STATES_ENUM.ERR_APPLIED].color }` : `bg-${ STATES[STATES_ENUM.UNKNOWN].color } bg-unmapped-state`;
+        }
+        if (row.bundles?.length && row.bundles?.some(bundle => bundle.state?.toLowerCase() === STATES_ENUM.NOT_READY)) {
+          return STATES[STATES_ENUM.NOT_READY].color ? `bg-${ STATES[STATES_ENUM.NOT_READY].color }` : `bg-${ STATES[STATES_ENUM.UNKNOWN].color } bg-unmapped-state`;
         }
 
-        return 'red-badge';
+        if (row.bundlesReady?.length === row.bundles?.length && row.bundlesReady && row.bundles?.length) {
+          return `bg-${ STATES[STATES_ENUM.ACTIVE].color }`;
+        }
+
+        return `bg-${ STATES[STATES_ENUM.NOT_READY].color } badge-class-default`;
       case 'resources':
+        if (row.status?.resources?.length && row.status?.resources?.every(resource => resource.state?.toLowerCase() === STATES_ENUM.ACTIVE)) {
+          return STATES[STATES_ENUM.ACTIVE].color ? `bg-${ STATES[STATES_ENUM.ACTIVE].color }` : `bg-${ STATES[STATES_ENUM.UNKNOWN].color } bg-unmapped-state`;
```

Comment:
@Shavindra I mapped it to the unknown state (so that we have a default and it doesn't break the UI visually) but I kept an extra class `bg-unmapped-state` so that we can catch problems while inspecting the page on the browser